cubed/storage/virtual.py

```diff
@@ -129,7 +130,7 @@ def __init__(
         self.chunks = template.chunks
         self.template = template
         if array.size > 0:
-            template[...] = array
+            template[...] = backend_array_to_numpy_array(array)
```

Comment:
FYI xarray has some more general versions of these types of array coercion functions here

https://github.com/pydata/xarray/blob/f343b7b23a275ebb17cdd91e73289525a627d39e/xarray/core/pycompat.py#L109